packages/zowe-explorer/src/dataset/DatasetTree.ts

```diff
@@ -1522,8 +1523,8 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
         }
     }
 
-    public async openWithEncoding(node: IZoweDatasetTreeNode): Promise<void> {
-        const encoding = await promptForEncoding(node);
+    public async openWithEncoding(node: IZoweDatasetTreeNode, encoding?: ZosEncoding): Promise<void> {
+        encoding = encoding ?? (await promptForEncoding(node));
```

Comment:
have you considered using `||=`? (e.g. `encoding ||= await prompt...`) 😋

Reply:
Typically I think we prefer using `??` over `||` - either one would probably work but the former has more strictly defined behavior. TIL that  [`??=`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) is a thing, so I'll use that 😋